packages/editor/src/components/post-trash/check.js

```diff
@@ -3,18 +3,27 @@
  */
 import { withSelect } from '@wordpress/data';
 
-function PostTrashCheck( { isNew, postId, children } ) {
-	if ( isNew || ! postId ) {
+function PostTrashCheck( { isNew, postId, canUserDelete, children } ) {
+	if ( isNew || ! postId || ! canUserDelete ) {
 		return null;
 	}
 
 	return children;
 }
 
 export default withSelect( ( select ) => {
-	const { isEditedPostNew, getCurrentPostId } = select( 'core/editor' );
+	const { isEditedPostNew, getCurrentPostId, getCurrentPostType } = select(
+		'core/editor'
+	);
+	const { getPostType, canUser } = select( 'core' );
+	const postId = getCurrentPostId();
+	const postType = getPostType( getCurrentPostType() );
+	const resource = postType?.[ 'rest_base' ] || '';
```

Comment:
Nit: I'd have avoided this temporary variable and just used postType in the check line 27.

Reply:
Hm, `resource` is used twice in line 27. Or do mean doing this instead?

`postId && postType ? canUser( 'delete', postType[ 'rest_base' ], postId ) : false,`

Reply:
yep, that's exactly what I meant. It's just a very small detail without importance though.